internal/daemon/daemon.go

```diff
@@ -351,9 +351,19 @@ func logit(handler http.Handler) http.Handler {
 	})
 }
 
+// Validate performs the checks needed to ensure that the Pebble daemon can be started.
+func (d *Daemon) Validate() error {
+	_, err := d.overlord.ServiceManager().Plan()
+	return err
+}
+
 // Init sets up the Daemon's internal workings.
 // Don't call more than once.
 func (d *Daemon) Init() error {
+	if err := d.Validate(); err != nil {
```

Comment:
If I read the code correctly, we will always perform all the validation steps twice for a normal run? It does not feel clean to me if I understand it correctly.

Reply:
@anpep!?

Reply:
That's correct. My opinion on that, as we discussed previously, is that a more elegant implementation would add too much complexity to the existing code, so I thought this is a good compromise. Will stay tuned to Gustavo's feedback once he reviews this PR. Thanks!